### PR TITLE
Allow JSONSelect to run in nodejs

### PR DESCRIPTION
**Pull request by tmpvar**
These commits include:
- check for exports, if available export JSONSelect instead of hanging it on the window
- a new test runner for node (src/test/run.js) which uses jsdom to bootstrap the doctest environment

To run the tests under node just use `node src/test/run.js`.  This command loads up each test html and executes a `window.doctest()`.  Since doctest logs results via console.log you can see them as the tests progress.

It would be really awesome if you could push this package into npm

### CODE REVIEW

File: src/jsonselect.js [hunk 1]
```diff
@@ -236,4 +235,4 @@
         forEach: function(sel, obj, fun) { return compile(sel).forEach(obj, fun) },
         compile: compile
     };
-})();
+})((typeof exports === "undefined") ? window : exports);
```
Comment (indutny) [inline src/jsonselect.js]: you can remove parentheses around <code>typeof exports === 'undefined'</code>, they ain't required

Comment (tmpvar) [inline src/jsonselect.js]: for you, 22f47851b6b9e168c552e4fca878d8fc3bebbb3c

**Comment by benatkin** [comment]
@tmpvar That's exactly what I thought should happen. I'm glad you did it! With @isaacs' distributed npm testing project the reference implementation will get CI for free. Mad props!

**Comment by benatkin** [comment]
On my [node branch](https://github.com/benatkin/JSONSelect/tree/node), I relaxed the node.js engine requirement to 0.4.7 (it still works for me) and made it so the tests can be run with `npm test`.

**Comment by lloyd** [comment]
@tmpvar Thanks!  I pulled in three of your commits unmolested, but decided to go a different route with testing.  the lex/parse/match tests using doctestjs were useful while writing the thing, but they really are redundant to the "conformance tests", which are explained here: https://github.com/lloyd/JSONSelect/blob/master/tests/README.md

Also moved node.js req to 0.4.7 as ben suggests.

`npm publish` -- check

One thing that bums me out is npm seems to be ignoring both .npmignore and top level 'files' entries on the package.  Just wanted to trim some fat.  whatever.

**Comment by isaacs** [comment]
Yep.  There are some bugs there.  1.0.7 coming soon (day or two) with some fixes.

**Comment by lloyd** [comment]
Thanks Isaac, I was sure I was Doing It Wrong.  I'll give 1.0.7 a try when it arrives!

**Pull request closed without merging**